contracts/BaseWrapper.sol

```diff
@@ -144,20 +156,23 @@ abstract contract BaseWrapper {
             availableShares = Math.min(availableShares, vaults[id].maxAvailableShares());
 
             if (availableShares > 0) {
-                // Compute amount to withdraw fully to satisfy the request
-                uint256 shares = amount
-                    .sub(withdrawn) // NOTE: Changes every iteration
-                    .mul(10**uint256(vaults[id].decimals()))
-                    .div(vaults[id].pricePerShare()); // NOTE: Every Vault is different
-
-                // Limit amount to withdraw to the maximum made available to this contract
-                shares = Math.min(shares, availableShares);
-
                 // Intermediate step to move shares to this contract before withdrawing
                 // NOTE: No need for share transfer if this contract is `sender`
-                if (sender != address(this)) vaults[id].transferFrom(sender, address(this), shares);
-
-                withdrawn = withdrawn.add(vaults[id].withdraw(shares, address(this)));
+                if (sender != address(this)) vaults[id].transferFrom(sender, address(this), availableShares);
+
+                if (amount < uint256(-1)) {
```

Comment:
can use the `type(uint256).max` from solidity 0.6.8 or use a constant to reference as well